maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java

```diff
@@ -87,22 +92,37 @@
     public PluginVersionResult resolve( PluginVersionRequest request )
         throws PluginVersionResolutionException
     {
+        ConcurrentMap<String, PluginVersionResult> cache = getCache( request.getRepositorySession().getData() );
+        String key = getKey( request );
```

Comment:
You don't need the key here, do you? Only when `result` is `null`. I would pull it down where it belongs.

Reply:
@michael-o you're right, this is fixed.  I've also aligned the PR on `master` to use a `ConcurrentMap` too (this was required to support JDK 1.7).